node/src/accountant/db_access_objects/pending_payable_dao.rs

```diff
@@ -403,21 +417,27 @@ mod tests {
 
         let result = subject.fingerprints_rowids(&[hash_1, hash_2]);
 
-        let first_expected_pair = (Some(1), hash_1);
+        let first_expected_pair = &(1, hash_1);
         assert!(
-            result.contains(&first_expected_pair),
+            result
+                .rowid_results
+                .as_slice()
+                .contains(first_expected_pair),
             "Returned rowid pairs should have contained {:?} but all it did is {:?}",
             first_expected_pair,
-            result
+            result.rowid_results.as_slice()
         );
-        let second_expected_pair = (Some(2), hash_2);
+        let second_expected_pair = &(2, hash_2);
         assert!(
-            result.contains(&second_expected_pair),
+            result
+                .rowid_results
+                .as_slice()
+                .contains(second_expected_pair),
             "Returned rowid pairs should have contained {:?} but all it did is {:?}",
             second_expected_pair,
-            result
+            result.rowid_results.as_slice()
         );
-        assert_eq!(result.len(), 2);
+        assert_eq!(result.rowid_results.as_slice().len(), 2);
```

Comment:
I'm sure you can call len() directly on the vector. Leave out the as_slice() fn and you can also check out the combination with .contains() you use above. While the first mentioned is a sure thing, the second is less. But please have a look.